File: pylox/pylox/Token.py
from TokenTypes import TokenType
from typing import Final


class Token:
    def __init__(
        self,
        type: TokenType,
        lexeme: str,
        literal: object,
        line: int,
    ) -> None:
        self.type: Final[TokenType] = type
        self.lexeme: Final[str] = lexeme
        self.literal: Final[object] = literal
        self.line: Final[int] = line

    def __str__(self) -> str:
        return f"{self.type} {self.lexeme} {self.literal}"

File: golox/token.go
package main

import "fmt"

type Token struct{
    tipe TokenType
    lexeme string
    literal interface{}
    line int
}

func NewToken(tipe TokenType, lexeme string, literal interface{}, line int) *Token{
    return &Token{
        tipe: tipe,
        lexeme: lexeme,
        literal: literal,
        line: line,
    }
}

func (token Token) String() string {
    return fmt.Sprintf("%v %v %v", token.tipe, token.lexeme, token.literal)
}
